src/lib.ml

```diff
@@ -1667,12 +1667,13 @@ module DB = struct
       Compile.for_lib t lib
 
   let resolve_user_written_deps_for_exes t exes
-        ?(allow_overlaps=false) deps ~pps ~variants =
+        ?(allow_overlaps=false) deps ~pps ~variants
+        ~optional =
     let lib_deps_info =
       Compile.make_lib_deps_info
         ~user_written_deps:deps
         ~pps
-        ~kind:Required
+        ~kind:(if optional then Optional else Required)
```

Comment:
Should we just make `optional` this variant rather than converting between the boolean?

This is not necessary to change in this PR, I'm just asking for an opinion.